File: C++/2358. Maximum Number of Groups Entering a Competition.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
 public:
  int maximumGroups(vector<int>& grades) {
    int cur = 0;
    int h = 0;
    while (cur + (h + 1) <= grades.size()) {
      cur += h + 1;
      h++;
    }
    return h;
  }
};

File: JavaScript/2358. Maximum Number of Groups Entering a Competition.js
/**
 * @param {number[]} grades
 * @return {number}
 */
var maximumGroups = function (grades) {
  let cur = 0;
  let h = 0;
  while (cur + h + 1 <= grades.length) {
    cur += h + 1;
    h++;
  }
  return h;
};
